tv_apps/tv_shared/js/card_manager.js

```diff
@@ -421,6 +421,18 @@
       }
     },
 
+    updateCard: function cm_updateCard(item) {
+      var that = this;
+      var index = (typeof item === 'number') ?
+        item : this._cardList.indexOf(item);
+
+      if (index >= 0) {
+        this.writeCardlistInCardStore().then(function() {
```

Comment:
I think you need to touch `this._cardList` before calling `this.writeCardlistInCardStore()`, otherwise the updated card will not get written into datastore.